crates/turborepo-auth/src/auth/mod.rs

```diff
@@ -5,3 +5,35 @@ mod sso;
 pub use login::*;
 pub use logout::*;
 pub use sso::*;
+use turborepo_api_client::Client;
+
+const VERCEL_TOKEN_DIR: &str = "com.vercel.cli";
+const VERCEL_TOKEN_FILE: &str = "auth.json";
+
+async fn extract_vercel_token(api_client: &impl Client) -> Result<Token, Error> {
+    let vercel_config_dir = turborepo_dirs::vercel_config_dir().ok_or_else(|| {
+        Error::Io(std::io::Error::new(
+            std::io::ErrorKind::NotFound,
+            "Vercel config directory not found",
+        ))
+    })?;
+
+    let vercel_token_path = vercel_config_dir
+        .join(VERCEL_TOKEN_DIR)
+        .join(VERCEL_TOKEN_FILE);
+    let contents = std::fs::read_to_string(&vercel_token_path)?;
+
+    #[derive(serde::Deserialize)]
+    struct VercelToken {
+        // This isn't actually dead code, it's used by serde to deserialize the JSON.
+        #[allow(dead_code)]
+        token: String,
+    }
+
+    let token = serde_json::from_str::<VercelToken>(&contents)?.token;
+    api_client.get_user(&token).await?;
+    Ok(Token {
+        token,
+        is_existing: true,
+    })
+}
```

Comment:
Sorry, should've been more clear about refactoring suggestions.

This would let us share the "display a token was found" code as they both operate on just having a token.

```suggestion
async fn extract_vercel_token() -> Result<String, Error> {
    let vercel_config_dir = turborepo_dirs::vercel_config_dir().ok_or_else(|| {
        Error::Io(std::io::Error::new(
            std::io::ErrorKind::NotFound,
            "Vercel config directory not found",
        ))
    })?;

    let vercel_token_path = vercel_config_dir
        .join(VERCEL_TOKEN_DIR)
        .join(VERCEL_TOKEN_FILE);
    let contents = std::fs::read_to_string(&vercel_token_path)?;

    #[derive(serde::Deserialize)]
    struct VercelToken {
        // This isn't actually dead code, it's used by serde to deserialize the JSON.
        #[allow(dead_code)]
        token: String,
    }

    serde_json::from_str::<VercelToken>(&contents)?.token
}
```